db/schema_tables.cc

```diff
@@ -879,9 +879,10 @@ future<> update_schema_version_and_announce(sharded<db::system_keyspace>& sys_ks
 future<> merge_schema(sharded<db::system_keyspace>& sys_ks, distributed<service::storage_proxy>& proxy, gms::feature_service& feat, std::vector<mutation> mutations)
 {
     co_await with_merge_lock([&] () mutable -> future<> {
-        bool flush_schema = proxy.local().get_db().local().get_config().flush_schema_tables_after_modification();
-        co_await do_merge_schema(proxy, std::move(mutations), flush_schema);
-        co_await update_schema_version_and_announce(sys_ks, proxy, feat.cluster_schema_features());
+        co_await smp::submit_to(0, [&] () mutable -> future<> {
```

Comment:
Isn't it better better to jump to shard zero before calling `with_merge_lock`? Otherwise you jump there 3 times.

Reply:
It is slightly better, yes.